Source/WebCore/platform/graphics/cocoa/VideoMediaSampleRenderer.mm

```diff
@@ -414,16 +414,19 @@ - (void)resetUpcomingSampleBufferPresentationTimeExpectations;
     return m_displayLayer.get();
 }
 
-RetainPtr<CVPixelBufferRef> VideoMediaSampleRenderer::copyDisplayedPixelBuffer()
+auto VideoMediaSampleRenderer::copyDisplayedPixelBuffer() -> DisplayedPixelBufferEntry
 {
     assertIsMainThread();
 
 #if HAVE(AVSAMPLEBUFFERDISPLAYLAYER_COPYDISPLAYEDPIXELBUFFER)
-    if (auto buffer = adoptCF([renderer() copyDisplayedPixelBuffer]))
-        return buffer;
+    if (!m_decompressionSession) {
+        if (auto buffer = adoptCF([renderer() copyDisplayedPixelBuffer]))
+            return { WTFMove(buffer), MediaTime::invalidTime() };
```

Comment:
shouldn't we return the wall time instead?

Reply:
No.

Reply:
isn't the currentTime a better estimation than nothing ?